src/Microsoft.Azure.EventHubs.Processor/AzureStorageCheckpointLeaseManager.cs

```diff
@@ -34,6 +37,14 @@ internal AzureStorageCheckpointLeaseManager(string storageConnectionString, stri
                 throw new ArgumentNullException(nameof(storageConnectionString));
             }
 
+            // Validate lease container name.
+            if (!Regex.IsMatch(leaseContainerName, @"^[a-z0-9](([a-z0-9\-[^\-])){1,61}[a-z0-9]$"))
+            {
+                throw new ArgumentException(
+                    "Lease container name is invalid. Please check naming conventions at https://msdn.microsoft.com/en-us/library/azure/dd135715.aspx",
```

Comment:
Can we say something about storage in here? And maybe something about this being based on the entity name? Because the container can be based off the entity name, correct?